### PR TITLE
fixing #52 hermes client retry on exception

### PR DESCRIPTION
**Pull request by dankraw**
Hermes client retrying on sender exception + response refactor (builder / api).

### CODE REVIEW

**Comment by adamdubiel** [comment]
:+1:

File: hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBuilder.java [hunk 1]
```diff
@@ -24,6 +21,9 @@ public static HermesClientBuilder hermesClient(HermesSender sender) {
     }
 
     public HermesClient build() {
+        if (retryCondition == null) {
+            retryCondition = new HermesClientBasicRetryCondition();
```
Comment (druminski) [inline hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBuilder.java]: You can avoid this null check through initialization in the field declaration.

Comment (dankraw) [inline hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBuilder.java]: ok, fixed

**Comment by druminski** [comment]
:+1:

**Pull request merged**